### PR TITLE
Fix task_value() with @copy_params and improve tests

### PR DESCRIPTION
**Pull request by adrien-berchet**

### CODE REVIEW

File: luigi_tools/task.py [hunk 1]
```diff
@@ -282,6 +282,18 @@ def _warn_on_wrong_global_param_type(_self, global_param_name, global_param_valu
...
+                # Fix task_value() to not normalize when the parameter does not have default value
+                def task_value(_self, task_name, param_name):
+                    # pylint: disable=protected-access
+                    value = _self._get_value(task_name, param_name)
+                    if value == _no_default_value:
+                        return value
+                    else:
+                        return _self._actual_task_value(task_name, param_name)
+
+                new_param._actual_task_value = new_param.task_value
```
Comment (anilbey) [inline luigi_tools/task.py:294]: I don't know if this statement is copying the new_param.task_value or just adding a reference to it. Just FYI.

Comment (adrien-berchet) [reply luigi_tools/task.py:294]: As far as I know it just adds a reference to the `task_value` method, which is what I need here so I can use it at line 292.

**Review by anilbey — APPROVED** [review]
Looks good to me.

File: luigi_tools/task.py (hunk 1 above)
Comment (anilbey) [reply luigi_tools/task.py:294]: it's ok as long as that's the desired behaviour.

Comment (adrien-berchet) [reply luigi_tools/task.py:294]: :+1:

**Pull request merged**